lib/renderer/init.js

```diff
@@ -39,10 +39,16 @@ electron.ipcRenderer.on('ELECTRON_INTERNAL_RENDERER_SYNC_WEB_FRAME_METHOD', (eve
 
 electron.ipcRenderer.on('ELECTRON_INTERNAL_RENDERER_ASYNC_WEB_FRAME_METHOD', (event, requestId, method, args) => {
   const responseCallback = function (result) {
-    event.sender.send(`ELECTRON_INTERNAL_BROWSER_ASYNC_WEB_FRAME_RESPONSE_${requestId}`, result)
+    Promise.resolve(result)
+      .then((resolvedResult) => {
+        event.sender.send(`ELECTRON_INTERNAL_BROWSER_ASYNC_WEB_FRAME_RESPONSE_${requestId}`, resolvedResult)
+      })
+      .catch((resolvedError) => {
+        console.error(`An async web frame method (${method}) returned a promise that threw an error: `, resolvedError)
```

Comment:
@paulcbetts I just modified the code so that the callback syntax remains the same but the `executeJavaScript` method now returns a promise that resolves the the result and rejects with any thrown error.  Let me know if that works for you 👍